src/stc/PlatWX.cpp

```diff
@@ -2414,10 +2417,47 @@ void wxSTCListBoxVisualData::RegisterImage(int type, const wxBitmap& bmp)
         return;
 
     ImgList::iterator it=m_imgList.find(type);
+    bool preExisting = false;
     if ( it != m_imgList.end() )
+    {
         m_imgList.erase(it);
+        preExisting = true;
+    }
 
     m_imgList[type] = bmp;
+
+    if ( preExisting )
+    {
+        m_imgAreaSize.Set(0,0);
+
+        for ( ImgList::iterator imgIt = m_imgList.begin() ;
+              imgIt != m_imgList.end() ; ++imgIt )
+        {
+            wxBitmap curBmp = imgIt->second;
+
+            if ( curBmp.GetWidth() > m_imgAreaSize.GetWidth() )
+            {
+                m_imgAreaSize.SetWidth(curBmp.GetWidth());
+            }
+
+            if ( curBmp.GetHeight() > m_imgAreaSize.GetHeight() )
+            {
+                m_imgAreaSize.SetHeight(curBmp.GetHeight());
+            }
+        }
+    }
+    else
+    {
+        if ( bmp.GetWidth() > m_imgAreaSize.GetWidth() )
+        {
+            m_imgAreaSize.SetWidth(bmp.GetWidth());
+        }
+
+        if ( bmp.GetHeight() > m_imgAreaSize.GetHeight() )
+        {
+            m_imgAreaSize.SetHeight(bmp.GetHeight());
+        }
```

Comment:
Use build-in `wxSize` function:
`m_imgAreaSize.IncTo(bmp.GetSize());`

Reply:
I made these changes.  I also realized it's only necessary to recheck the entire list of images if an item is being replaced AND the new item has a different size.